### DISCUSSION TITLE
Export types from @medusajs/medusa-js

### DISCUSSION THREAD

**Discussion by ghost** [medusajs/medusa#4942] [Feature Requests]
Medusa has been really useful for us where I work, and we integrate our backend with our Medusa instance via the Medusa JS SDK. However, we have found that in order to get TypeScript typing for the interfaces we send and receive with the SDK, we need to install the `@medusajs/medusa` module, which feels like reinventing the wheel given that that module includes a copy of the whole Medusa backend and adds dozens of additional dependencies to our backend. Installing the `medusa-interfaces` module also doesn't work, as that module seems to only export types for services rather than for the entity types Medusa exposes.

I've already seen the RFC on #558, and the previous issue #3082, but both of these are quite old and seem to have been resolved, which was what drove me to make a new feature request. Let me know if there is an existing discussion on this and I will happily contribute to that instead :)

**Comment by pevey** [comment]
As the new module architecture is released bit by bit (Products module released so far), the associated types are being moved to the @medusajs/types package. Eventually, they will will all be there.

**Reply by olivermrbl** [reply]
@barnabycollins-s2z depending on your usage, you might also want to check out `@medusajs/client-types`

**Reply by ghost** [reply]
@olivermrbl - apologies; been quite busy lately! The `@medusajs/client-types` module is exactly the sort of thing I was looking for; thank you :)

Is your plan to transition from `@medusajs/client-types` to `@medusajs/types`, or do the two modules serve slightly different purposes?
